domain/ide/workspace/folder.go

```diff
@@ -156,33 +178,95 @@ func (f *Folder) DocumentDiagnosticsFromCache(file string) []snyk.Issue {
 	return issues
 }
 
-func (f *Folder) processResults(product product.Product, issues []snyk.Issue, err error) {
-	if err != nil {
-		f.scanNotifier.SendError(product, f.path)
-		log.Err(err).Str("method", "processResults").
-			Msgf("%s returned an error: %s", product, err.Error())
+func (f *Folder) processResults(scanData snyk.ScanData) {
+	if scanData.Err != nil {
+		f.scanNotifier.SendError(scanData.Product, f.path)
+		log.Err(scanData.Err).Str("method", "processResults").
+			Msgf("%s returned an error: %s", scanData.Product, scanData.Err.Error())
 		return
 	}
 
 	dedupMap := f.createDedupMap()
 
 	// TODO: perform issue diffing (current <-> newly reported)
 	// Update diagnostic cache
-	for _, issue := range issues {
+	for _, issue := range scanData.Issues {
 		cachedIssues, _ := f.documentDiagnosticCache.Load(issue.AffectedFilePath)
 		if cachedIssues == nil {
 			cachedIssues = []snyk.Issue{}
 		}
 
 		if !dedupMap[f.getUniqueIssueID(issue)] {
 			cachedIssues = append(cachedIssues, issue)
+			switch issue.Severity {
+			case snyk.Critical:
+				scanData.Critical++
+			case snyk.High:
+				scanData.High++
+			case snyk.Medium:
+				scanData.Medium++
+			case snyk.Low:
+				scanData.Low++
+			}
```

Comment:
bug: we should count by product. This is going to be addressed in a follow up PR